machine/v1beta1/types_awsprovider.go

```diff
@@ -107,6 +107,22 @@ type AWSMachineProviderConfig struct {
 	// If this value is selected, capacityReservationID must be specified to identify the target reservation.
 	// +optional
 	MarketType MarketType `json:"marketType,omitempty"`
+
+	// hostID specifies the Dedicated Host on which the instance must be started.
+	// This field is mutually exclusive with DynamicHostAllocation.
+	// +kubebuilder:validation:Pattern=`^h-[0-9a-f]{17}$`
+	// +kubebuilder:validation:MaxLength=19
+	// +openshift:enable:FeatureGate=AWSDedicatedHosts
+	// +optional
+	HostID *string `json:"hostID,omitempty"`
```

Comment:
What is the difference between setting this to `""` and omitting the field entirely?

Reply:
If there is no difference, this should not be a pointer and should have a minimum length of 1. This is probably what the linter is complaining about.

Reply:
Ah, I went into standard API review mode here and forgot this API is webhook validation :facepalm:

Thanks for catching that!

>We may want to even disable the serialization rules on these particular APIs somehow

Can we do this via codegen configurations?

Reply:
> Can we do this via codegen configurations?

No, but we should be able to disable using the `.golangci-lint.yaml` config, ideally we could have a different config for the APIs that act like this, these MAPI ones aren't the only ones (e.g. the aggregated APIs we support too)